modules/exploits/linux/http/mobileiron_core_log4shell.rb

```diff
@@ -0,0 +1,133 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::Log4Shell
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::Remote::CheckModule
```

Comment:
You have a check method defined, and the check module targets header-based injections so it should be fine to remove this.